INTER-235-MobileMenuComponent-모바일메뉴 공용 컴포넌트 구현

mulgko opened the pull request:
PLUG-01-Button -> dev

### PR 타입(하나 이상의 PR 타입을 선택해주세요)

- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트
- [ ] 코드 리펙토링

### 작업 내용

- 모바일 메뉴에서 보이는 메뉴 UI 개발
- 댓글이나 답글 달기를 눌렀을 때 해당 클릭지점에서 드롭다운 되는 방식이 아닌, 아래에서 스프링 되는 메뉴 
- drag(핀치)가 되므로 UI 개선이 이루어짐
 

### 스크린샷
<img width="392" alt="image" src="https://github.com/user-attachments/assets/05f69721-2574-45d6-bedd-81d212f70d1b" />


### 리뷰 요구사항

- case 형식으로 컴포넌트들을 나누어놓았습니다. 그렇다보니 variants가 중복으로 작성되는 경우가 생깁니다. 이를 최적화 할 수 있는 방법이 있을까요?

mulgko committed:
feat: Tag Component 생성
- Tailiwind varaints 설정
- 난이도 태그 생성
- 카테고리 태그 생성
- 카드 태그 생성

mulgko committed:
refactor: radio 기능 삭제
radio 태그 삭제

mulgko committed:
feat: 태그 추가 및 모바일 적용
폰트사이즈 조절
close tag 추가
정답, 오답 tag 추가
tag 레이아웃 조절
props로 data 추가

mulgko committed:
test: test

mulgko committed:
feat: 모바일 메뉴 컴포넌트
시작

mulgko committed:
feat: 모바일 메뉴 생성

mulgko committed:
fixed: 폰트 변수 변경
text => font #11

mulgko committed:
feat: mobile menu
other user compoonent
current user component

mulgko committed:
feat: menu 추가
- case문으로 다양한 컴포넌트 렌더링
- setting, share.. 등등

mulgko committed:
refactor: 리팩토링

mulgko committed:
Merge remote-tracking branch 'origin/ISSUE-26-ButtonDefaultDelete' into INTER-235-MobileMenuComponent

mulgko committed:
refactor: 코드 리팩토링

GiToon10100011 commented on src/app/components/tag/Tag.tsx:
MobileMenu 컴포넌트 PR인데 Tag가 왜 들어가.. 있나요..?

mulgko replied:
오우.. Tag페이지에서 create한 것 같습니다

GiToon10100011 commented on src/app/components/input/input.variants.ts:
```diff
@@ -41,4 +41,10 @@ export const labelVariants = tv({
       true: "text-red-500",
     },
   },
+<<<<<<< HEAD
```
input이 merge가 다 안된채로 PR이 올라간것 같습니다..

GiToon10100011 commented on src/app/components/mobilemenu/MobileMenu.tsx:
```diff
@@ -0,0 +1,162 @@
+  const y = useMotionValue(0);
+
+  // Animation variants
+  const menuVariants = {
```
variants.ts로 빼주세요.

GiToon10100011 commented on src/app/components/mobilemenu/MobileMenu.tsx:
```diff
@@ -0,0 +1,162 @@
+  );
+};
+
+export default MobileMenu;
```
저번 PR에서도 말했듯이 default는 테스트 페이지를 제외하고는 사용하지 않습니다. export const로 수정해주세요.

GiToon10100011 commented on src/app/components/mobilemenu/mobile.menu.variants.ts:
```diff
@@ -0,0 +1,30 @@
+import { tv } from "tailwind-variants";
+
+export const contentVariants = tv({
```
variants를 여러개 만들기 보다는 slots사용을 권장하겠습니다.

mulgko committed:
Merge remote-tracking branch 'origin/dev' into INTER-235-MobileMenuComponent

mulgko committed:
reafctor: 코드 리팩토링
- tag 삭제
- input merge
- animation variants 생성
- export const로 수정
- ᆫ slot 사용

GiToon10100011 commented on src/app/components/mobilemenu/mobile.menu.variants.ts:
사소하지만 파일 컨벤션에 맞게 mobileMenu.variants.ts로 수정 부탁드립니다. 폴더도 mobilemenu가 아니라 camelCase에 맞게 mobileMenu로 수정부탁드립니다.

mulgko replied:
넵!

GiToon10100011 replied:
아직 폴더와 type.ts가 수정이 안된 것 같습니다..!

GiToon10100011 commented on src/app/components/mobilemenu/mobile.menu.variants.ts:
어차피 mobileMenu는 이름 그대로, 모바일에서만 사용될 메뉴인데 따로 menuVariants를 만드신 이유가 궁금합니다!

mulgko replied:
기능 분리를 위해서 mobileMenuVariants와 menuItemVariants 이렇게 두 개로 나누었습니다. 

![image](https://github.com/user-attachments/assets/3daf7a97-4cbd-460e-81e4-4d6a904f8efb)
이렇게 하나도 묶었을 때 type: danger이 적용이 되어야하는데,
꼭 이렇게 분리를 해야 적용이 되는 문제가 있습니다.

![image](https://github.com/user-attachments/assets/846dfb00-a249-4252-b978-c50491f2b299)
<img width="381" alt="image" src="https://github.com/user-attachments/assets/2b4c5504-b5af-4854-a848-fc45a64f47c0" />

GiToon10100011 replied:
```ts
export const mobileMenuVariants = tv({
  slots: {
    mobileMenu: "fixed bottom-0 left-0 right-0 bg-white rounded-t-3xl z-50",
    content: "flex flex-col gap-4",
    backdrop: "fixed inset-0 bg-black/50 z-40",
    handle: "w-12 h-1 bg-gray-300 rounded-full mx-auto mb-5",
    menuItem: "block py-3 font-regular-18 flex items-center gap-2 ",
  },
  variants: {
    type: {
      default: {
        menuItem: "",
      },
      danger: {
        menuItem: "text-incorrect/60",
      },
    },
  },
});
```

다음과 같이 선언 후, 

```ts
className={mobileMenuVariants({ type: 'danger' }).menuItem()}

//or
className={menuItem({ type: "default" })}
```

해당 방법으로 사용하면 적용 잘됩니다! slots랑 variants를 같이 적용시키려면 좀 특이한 구조를 따라야하더라고요

mulgko replied:
wow 해당 방법으로 하니까 잘 되네요 감사합니다 :)

mulgko committed:
fix: 폴더명 수정

mulgko committed:
fix: 폴더명 수정, variants 수정
- 폴더명 mobilemenu => mobileMenu
- meunItem() => mobileMenuVairants({type: danger}).menuItem()
